### DISCUSSION TITLE
[Bug]:  Windows build fails at openssl

### DISCUSSION THREAD

**Discussion by Wosler-Menna** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#2006] [Q&A] [answered]
### Please confirm you have already done the following

- [X] I have searched the repository for related/existing bug reports
- [X] I have all the details the issue requires

### Please answer the following prompt

- [X] This issue is replicable using the unmodified sample application

### Describe the bug

When I try to run the build_windows_openssl.bat script, it fails at 11% during the update step for 'project_libopenssl'. I followed the steps in [configuring for windows](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c?tab=readme-ov-file#configuring-on-windows). 

### Expected Behavior

Expected to build the sdk without issues

### Current Behavior

The build process fails with the following error:
```

[ 11%] Performing update step for 'project_libopenssl'
CMake Error at C:/code/amazon-kinesis-video-streams-webrtc-sdk-c/open-source/libopenssl/build/tmp/project_libopenssl-gitupdate.cmake:25 (message):
  Failed to get the hash for HEAD:

  fatal: not a git repository: '.git'


NMAKE : fatal error U1077: 'echo >nul && "C:\Program Files\CMake\bin\cmake.exe" -P C:/code/amazon-kinesis-video-streams-webrtc-sdk-c/open-source/libopenssl/build/tmp/project_libopenssl-gitupdate.cmake' : return code '0x1'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.40.33807\bin\HostX86\x64\nmake.exe" -s -f CMakeFiles\project_libopenssl.dir\build.make /nologo -SL CMakeFiles\project_libopenssl.dir\build' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.40.33807\bin\HostX86\x64\nmake.exe" -s -f CMakeFiles\Makefile2 /nologo -LS all' : return code '0x2'
Stop.
CMake Error at CMake/Utilities.cmake:75 (message):
  CMake step for libopenssl failed: 2
Call Stack (most recent call first):
  CMakeLists.txt:157 (build_dependency 
```

### Reproduction Steps

1.  Follow windows configuration steps in the readme
1.  Run ` .github\build_windows_openssl.bat `

### WebRTC C SDK version being used

master branch, release 1.10 windows

### Compiler and Version used

Visual studio 2022

### Operating System and version

Windows 10 

### Platform being used

-

**Comment by niyatim23** [comment, marked as answer]
We have a successful openssl build in our CI [here](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/actions/runs/9010113338/job/24760863028). Please closely follow all the steps below and see if it helps. We have verified that the build and the same both work on Windows. This issue maybe specific to your setup
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/4f1d3add4c0f88212fe6646898e5aba140ecbb3c/.github/workflows/ci.yml#L536-L588

**Reply by Wosler-Menna** [reply]
Thank you, I retraced all the steps and it seems to work now
